### PR TITLE
prerequisite: add spdk environment setup manifest

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#5738

### CODE REVIEW

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Just one suggestion. 

BTW, where to assign CPU core to SPDK tgt? at runtime?

File: deploy/prerequisite/longhorn-spdk-setup.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: longhorn-spdk-setup
+  labels:
+    app: longhorn-spdk-setup
+  annotations:
+    command: &cmd rm -rf ${SPDK_DIR}; git clone -b longhorn https://github.com/longhorn/spdk.git ${SPDK_DIR} && bash ${SPDK_DIR}/scripts/setup.sh; if [ $? -eq 0 ]; then echo "SPDK environment is configured successfully"; else echo "Failed to configure SPDK environment error code $?"; fi; rm -rf ${SPDK_DIR}
```
[the remark this answers is not in the document]
Comment (innobead) [reply deploy/prerequisite/longhorn-spdk-setup.yaml:8]: Did you forget to force push :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (derekbit) [reply deploy/prerequisite/longhorn-spdk-setup.yaml:8]: > Can we add a first argument for setup.sh via env var? 

Do you mean don't use the envs?

File: deploy/prerequisite/longhorn-spdk-setup.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          - -c
+          - *cmd
+        image: alpine:3.12
+        env:
+        - name: SPDK_DIR
+          value: /tmp/spdk
+        - name: HUGEMEM
+          value: "2048"
+        - name: DRIVER_OVERRIDE
+          value: "uio_pci_generic"
```
[the remark this answers is not in the document]
Comment (innobead) [reply deploy/prerequisite/longhorn-spdk-setup.yaml:36]: I mean below. WDYT?

```bash
bash ${SPDK_DIR}/scripts/setup.sh ${SETUP_CMD}
```

the cmd can be config by default, so we can run it with different commands later on for debugging purposes.

[the remark this answers is not in the document]
Comment (derekbit) [reply deploy/prerequisite/longhorn-spdk-setup.yaml:36]: Got it.

**Comment by derekbit** [comment]
> In general, LGTM. Just one suggestion.
> 
> BTW, where to assign CPU core to SPDK tgt? at runtime?

Yes, configured at runtime when starting spdk_tgt.
```
instance-manager-8c4051a90e8a9c22a4d64744b6afdc34:/ # spdk_tgt -h
spdk_tgt [options]
options:
 -c, --config <config>     JSON config file (default none)
     --json <config>       JSON config file (default none)
     --json-ignore-init-errors
                           don't exit on invalid config entry
 -d, --limit-coredump      do not set max coredump size to RLIM_INFINITY
 -g, --single-file-segments
                           force creating just one hugetlbfs file
 -h, --help                show this usage
 -i, --shm-id <id>         shared memory ID (optional)
 -m, --cpumask <mask or list>    core mask (like 0xF) or core list of '[]' embraced (like [0,1,10]) for DPDK
 -n, --mem-channels <num>  channel number of memory channels used for DPDK
 -p, --main-core <id>      main (primary) core for DPDK
 -r, --rpc-socket <path>   RPC listen address (default /var/tmp/spdk.sock)
 -s, --mem-size <size>     memory size in MB for DPDK (default: 0MB)
     --disable-cpumask-locks    Disable CPU core lock files.
     --silence-noticelog   disable notice level logging to stderr
     --msg-mempool-size <size>  global message memory pool size in count (default: 262143)
 -u, --no-pci              disable PCI access
     --wait-for-rpc        wait for RPCs to initialize subsystems
     --max-delay <num>     maximum reactor delay (in microseconds)
 -B, --pci-blocked <bdf>
 ```

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
Awesome @derekbit 👏

**Pull request merged**